go/vt/sqlparser/constants.go

```diff
@@ -1070,3 +1070,12 @@ const (
 	IndexTypeSpatial
 	IndexTypeFullText
 )
+
+// FkChecksState is the enum to store the value of `foreign_key_checks` MySQL variable.
+type FkChecksState int
+
+const (
+	FkChecksUnspecified FkChecksState = iota
+	FkChecksOn
+	FkChecksOff
```

Comment:
We need to document clearly what's the semantic difference between foreign checks being unspecified and foreign checks being off. I don't see the difference by looking at the code. If there is no meaningful difference, `FkChecksOff` should be the `iota` instead of having 3 states.

Reply:
Looking at the places where this is used, this doesn't seem like the proper design at all. The `FkCheckState` is not part of the AST, it only exists as part of custom syntax in a _comment_. I don't think the `sqlparser` package should export this type.

If the "unspecified" case must be distinct to set, you can model this ternary type as `*bool` like ProtoBuf does. I think that's plenty idiomatic for Go. If there's no difference between `Unspecified` and `Off`, then a simple `bool` will do the same job.

Reply:
unspecified is the case that we do not set anything (default keyspace level setting is used)
off and on change the way planning works and override the defaults of vtgate

Reply:
I have used `*bool` to store the foreign key checks state now.